src/qibolab/designs/abstract.py

```diff
@@ -0,0 +1,50 @@
+from abc import ABC, abstractmethod
+
+
+class AbstractInstrumentDesign(ABC):
```

Comment:
Can you add an extended comment on how is supposed to be the overall workflow, either in the class or module level docstring?

Like:
- suppose I have a concrete `InstrumentDesign`, made of this few components
- you start by `connect`ing
- then you can assign qubits with `setup`
- after that you `start` and you can
  - `play` any number of sequences, by repeated (sequential) invocations
  - `stop` at any time
- once stopped you can
  - resume, by `start`ing again
  - eventually `disconnect`

In particular, if you like it, I would use mermaid format https://mermaid.js.org/ to make a flowchart. There is also a [Sphinx integration](https://github.com/mgaitan/sphinxcontrib-mermaid) to render the diagrams.

Reply:
An advantage of Mermaid is that it is also supported in GitHub's comments:
```mermaid
graph TD
inst[<pre>InstrumentDesign</pre>]
connected[connect, w or w/o qubits]
inst -- <pre>connect</pre> --> connected
connected -. <pre>setup</pre> .-> connected
connected -- <pre>start</pre> --> running
running -- <pre>stop</pre> --> connected
running -- <pre>play</pre> --> running
connected -- <pre>disconnect</pre> --> inst
```